### PR TITLE
Pre-commit - code style/quality

### PR DESCRIPTION
**Pull request by diegorafrh**
Added *pre-commit* to help with the code quality/style.

Pre-commit runs a set of code analysis packages/hooks, such as:

- black
- flake8
- reorder-python-imports
- trailing-whitespace
- end-of-file-fixer
- check-yaml
- debug-statements
- pyupgrade

You can check the hooks available here: https://github.com/pre-commit/pre-commit-hooks

As you can see a lot of files has been automatically modified by **black** and I had to adjust manually a few others to fix all **flake8** warnings.

This PR comes from the **poc/ci** , I decided to put it apart so we can still merge the **poc/ci** without all this stuff and let this open to discuss and adjust the **pre-commit** till it fits the project needs.

Install **pre-commit**: `pipenv install pre-commit`
Running: `pre-commit run --all` 

It is worth mention that this effort is tied with the CI process to improve our Quality Gate

### CODE REVIEW

File: app/auth/__init__.py [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
-__all__ = ["current_identity",
-           "bearer_token_handler",
-           "authentication_header_handler"]
+__all__ = ["current_identity", "bearer_token_handler", "authentication_header_handler"]
 
 logger = get_logger(__name__)
 
 
-def authentication_header_handler(apikey, required_scopes=None):
+def authentication_header_handler(apikey, required_scopes=None):  # noqa
```
Comment (catastrophe-brandon) [inline app/auth/__init__.py]: Any particular reason for adding `#noqa` here?

Comment (diegorafrh) [reply app/auth/__init__.py]: AFAIK Connexion injects this kwarg by default, so we need to keep it in the function signature.
Here's an example of its usage https://github.com/zalando/connexion/blob/master/examples/swagger2/basicauth/app.py

**Comment by catastrophe-brandon** [comment]
To help keep the history clean, would it be possible to squash this before merging (or as part of the merge operation)?

**Comment by Glutexo** [comment]
My original idea was to do this in small steps, fixing only a few things at once. Advantages would be that

- the pull requests would be easily reviewable,
- it’d be clear that nothing is getting broken,
- arising conflicts in other pull request could be easily resolved.

Also it’s not 100% necessary to do to fixes and add the pre-commit hook at the same time. We can softly enforce it for the subsequent PRs and once everything is fixed, we can enable the hooks.

**Comment by catastrophe-brandon** [comment]
> My original idea was to do this in small steps, fixing only a few things at once. Advantages would be that
> 
> * the pull requests would be easily reviewable,
> * it’d be clear that nothing is getting broken,
> * arising conflicts in other pull request could be easily resolved.
> 
> Also it’s not 100% necessary to do to fixes and add the pre-commit hook at the same time. We can softly enforce it for the subsequent PRs and once everything is fixed, we can enable the hooks.

^ Good idea. I fully support this.

**Comment by diegorafrh** [comment]
We definitely should adjust the service's code in baby steps that's why I have opened it as draft just to show you guys how the pre-commit is going to impact the code, especially because black adjusts the code itself.

I can pull it back and just leave the pre-commit/hooks file ready for use and you guys can decide when/how to run it against the code.

**Comment by Glutexo** [comment]
This is the current state of the code, as Flake8 says:

```
      count                                        description
code                                                          
E501    156                 line too long (82 > 79 characters)
E128     14  continuation line under-indented for visual in...
E302      6                    expected 2 blank lines, found 0
E265      3               block comment should start with ‘# ‘
E231      3          missing whitespace after ‘,’, ‘;’, or ‘:’
E305      2  expected 2 blank lines after end of function o...
E303      2                           too many blank lines (3)
W601      1                 .has_key() is deprecated, use ‘in’
W391      1                          blank line at end of file
W291      1                                trailing whitespace
E402      1             module level import not at top of file
E261      1          at least two spaces before inline comment
E127      1  continuation line over-indented for visual indent
```

Not great, not terrible.

**Comment by Glutexo** [comment]
Just linking an existing issue #189.

**Comment by dehort** [comment]
Does this PR need to be modified to only include the pre-commit hook changes?  Diego, is that something that you were planning on doing?

**Comment by diegorafrh** [comment]
> Does this PR need to be modified to only include the pre-commit hook changes? Diego, is that something that you were planning on doing?

Nope, this PR doesn't apply any longer. @Glutexo did an amazing job splitting all the changes in minor PRs instead of merging this huge one - we can close this

File: app/auth/__init__.py (hunk 1 above)
Comment (Glutexo) [reply app/auth/__init__.py]: My Flake8 doesn’t complain about this unused argument variable with default configuration.

**Pull request closed without merging**